finish migrating project to bevy 0.9.0 & cleanup dependencies

mirenbhakta opened the pull request:
Cargo.lock has no mention of previous versions of bevy now

mirenbhakta committed:
finish migrating project to bevy 0.9.0

mirenbhakta committed:
remove unnecessary Cargo.lock files within rc_protocol and world_generation crates

DarkZek commented on lib/rc_networking/src/server/mod.rs:
```diff
@@ -113,7 +115,7 @@ impl ServerSocket {
                         Protocol::Disconnect(Disconnect::new(0)),
                         UserId(0),
-                    ));
+                    )).expect("failed to send disconnect packet");
```
This was purposefully left without an unwrap because if the users socket has already shut and the tokio threads despawned, then the reciever for this is gone and it will fail. We don't really care if it failed to send a disconnect, so it shouldn't panic the server if it did fail

mirenbhakta replied:
The proper way to discard a Result is something like
`let _discard = `
which will not trigger the unused Result compiler lint.
I will make a commit to update this.

mirenbhakta committed:
clean up dependencies
remove unused tokio dependency in client
begin using workspace dependencies, starting with bevy and bincode
every crate now uses the bevy crate instead of its sub crates
fix imports due to the above change

mirenbhakta committed:
resolving https://github.com/DarkZek/RustCraft/pull/14#discussion_r1022176533
